charts/devtron/values.yaml

```diff
@@ -105,3 +104,151 @@ components:
         GIT_REPO_URL: "https://github.com/devtron-labs/devtron.git"
         DB_NAME: "casbin"
         GIT_HASH: "e9175e9845c9ca8f6260ea58ce7ad22118244eac"
+
+argo-cd:
+  enabled: false
+  fullnameOverride: "argocd"
+  configs:
+    secret:
+      createSecret: false
+  # argocd-application-controller
+  controller:
+    args:
+      # -- define the application controller `--status-processors`
+      statusProcessors: "50"
+      # -- define the application controller `--operation-processors`
+      operationProcessors: "25"
+      # -- define the application controller `--app-hard-resync`
+      appHardResyncPeriod: "0"
+      # -- define the application controller `--app-resync`
+      appResyncPeriod: "180"
+      # -- define the application controller `--self-heal-timeout-seconds`
+      selfHealTimeout: "5"
+      # -- define the application controller `--repo-server-timeout-seconds`
+      repoServerTimeoutSeconds: "200"
+    extraArgs:
+    - --kubectl-parallelism-limit
+    - "35"
+    containerSecurityContext:
+      allowPrivilegeEscalation: false
+      capabilities:
+        drop:
+        - all
+      readOnlyRootFilesystem: true
+      runAsNonRoot: true
+    env:
+    - name: ARGOCD_RECONCILIATION_TIMEOUT
+      valueFrom:
+        configMapKeyRef:
+          key: timeout.reconciliation
+          name: argocd-cm
+          optional: true
+    affinity:
+      podAntiAffinity:
+        preferredDuringSchedulingIgnoredDuringExecution:
+        - podAffinityTerm:
+            labelSelector:
+              matchLabels:
+                app.kubernetes.io/name: argocd-application-controller
+            topologyKey: kubernetes.io/hostname
+          weight: 100
+        - podAffinityTerm:
+            labelSelector:
+              matchLabels:
+                app.kubernetes.io/part-of: argocd
+            topologyKey: kubernetes.io/hostname
+          weight: 5
+  # argocd-dex-server
+  dex:
+    enabled: false
+  
+  redis:
+    affinity:
+      podAntiAffinity:
+        preferredDuringSchedulingIgnoredDuringExecution:
+        - podAffinityTerm:
+            labelSelector:
+              matchLabels:
+                app.kubernetes.io/name: argocd-redis
+            topologyKey: kubernetes.io/hostname
+          weight: 100
+        - podAffinityTerm:
+            labelSelector:
+              matchLabels:
+                app.kubernetes.io/part-of: argocd
+            topologyKey: kubernetes.io/hostname
+          weight: 5
+    
```

Comment:
Included comments for better readability

Reply:
Comments added